flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/api/TableEnvironmentTest.scala

```diff
@@ -706,6 +717,40 @@ class TableEnvironmentTest {
     assertFalse(tableEnv.listUserDefinedFunctions().contains("f2"))
   }
 
+  @Test
+  def testExecuteSqlWithCreateFunctionWithResource(): Unit = {
+    val userJar = UserClassLoaderJarTestUtils.createJarFile(
+      tempFolder.newFolder("test-jar"),
+      "test-classloader.jar",
+      udfClass,
+      udfCode)
+
+    val tableResult1 = tableEnv.executeSql(
```

Comment:
I think this test is not correct, the aim of this pr is to support register function by table api, so we should test register function by table api instead of sql. You should add tests in `FunctionITCase`, and test function work in streaming and batch mode, please refer to related test in `FunctionITCase`.